src/xscen/catalog.py

```diff
@@ -1018,7 +1021,7 @@ def unstack_id(df: Union[pd.DataFrame, ProjectCatalog, DataCatalog]) -> dict:
             [
                 col
                 for col in subset.columns
-                if bool(re.search(f"((_)|(^)){str(subset[col].iloc[0])}((_)|($))", ids))
+                if bool(re.search(f"((_)|(^)){subset[col].iloc[0]!s}((_)|($))", ids))
```

Comment:
Is that doing the right thing?

Reply:
Yes, this is proper using in f-strings.

See: https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals